src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/jira/home/SharedHomeFormula.kt

```diff
@@ -37,6 +47,31 @@ internal class SharedHomeFormula(
             it.safeExecute("sudo mv $jiraHome/logos $localSharedHome")
             ubuntu.install(it, listOf("nfs-kernel-server"))
             it.execute("sudo echo '$localSharedHome $localSubnet(rw,sync,no_subtree_check,no_root_squash)' | sudo tee -a /etc/exports")
+
+            if (s3StorageBucketName != null && storeInS3.isNotEmpty()) {
+                AwsCli(awsCliVersion).ensureAwsCli(it)
+                if (storeInS3.contains(ATTACHMENTS)) {
+                    // Copy the attachment data from NFS onto S3
+                    // Use xargs to split the work across 30 concurrent jobs.  This is much faster than a single job.
+                    it.safeExecute(
+                        cmd = "export AWS_RETRY_MODE=standard; export AWS_MAX_ATTEMPTS=10; cd $localSharedHome/data && ( find attachments -mindepth 1 -maxdepth 1 -type d -print0 | xargs -n1 -0 -P30 -I {} aws s3 cp --recursive --only-show-errors {}/ s3://$s3StorageBucketName/{}/ )",
```

Comment:
Are we planning to share this upload method with customers? In JPT we like to have references to documentation in the code, as we want the automation to simulate the admin.

Reply:
At the moment we will be recommending that customers use AWS DataSync instead of this approach.  If we do end up suggesting this approach to customers, then I'll be sure to update the code with a link to documentation.

Reply:
Would it be possible to use the AWS DataSync in our code? Our [approach](https://stash.atlassian.com/projects/JIRASERVER/repos/jira-performance-tests/browse/docs/APPROACH.md) was always to imitate admins, which has many benefits.

Reply:
I'm open to your thoughts here but I think using DataSync in JPT may be overkill.  DataSync is recommended to customers because it's very robust, and reliable for large instances (eg. 1TB+ of attachments).
To setup DataSync requires quite a lot of work and in general a customer with only a few GB (or 10's of GBs) of data like in the JPT datasets would use an approach like `aws cp`.  We actually expect some customers with smaller instances to use this approach (ie `aws cp`) instead of DataSync.  Perhaps this is more an issue with our documentation and we should look at recommending this approach there as well.

Reply:
We want our datasets in JPT to be of the scale of our customers.

If we want to recommend usage of `aws cp` to our customers then I think it should be fine for JPT as well. Let's just be consistent, because:
 * If we will have problems with the recommendation here, then our customers may also have those.
 * In case we will improve our recommendation we will also want to apply it in JPT to benefit.
 * With using different approach then recommendation we will face problems that our customers don't have.

Is there something problematic in DataSync that makes it hard to use in JPT specifically?

Reply:
I hear what you're saying around wanting to imitate Jira admins, it's a good form of added testing and understanding what our customers experience. At the moment, our documentation recommends DataSync, however it doesn't provide a definitive solution.  It leaves the decision of how to copy data to the admin and doesn't provide any specific commands.  Here's the relevant documentation:
![image](https://github.com/atlassian/aws-infrastructure/assets/11641643/a5ce307f-849d-49da-b40d-4c89c72abbd8).

It's definitely possible to implement DataSync, though it will require more setup and configuration. In addition it may be prohibitively slow for the JPT use case (ie take closer to 40 minutes to copy default dataset). To set it up, we'll need to create a DataSync agent (on a new EC2 instance), configure source and destination locations, and set up task execution.